### PR TITLE
Enable nullability in some ToolStripItem members

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some ToolStripItem members.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/8174)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripItem.ToolStripItemAccessibleObject.cs [hunk 1]
```diff
@@ -312,7 +312,7 @@ public override Rectangle Bounds
                     if (Owner.IsOnDropDown)
                     {
                         // Return the owner item as the accessible parent.
-                        ToolStripDropDown dropDown = Owner.GetCurrentParentDropDown();
+                        ToolStripDropDown dropDown = Owner.GetCurrentParentDropDown()!;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripItem.ToolStripItemAccessibleObject.cs:315]: How do we know this isn't null?

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripMenuItem.cs [hunk 2]
```diff
@@ -988,7 +988,7 @@ private void OnMouseButtonStateChange(MouseEventArgs e, bool isMouseDown)
             bool showDropDown = true;
             if (IsOnDropDown)
             {
-                ToolStripDropDown dropDown = GetCurrentParentDropDown() as ToolStripDropDown;
+                ToolStripDropDown dropDown = GetCurrentParentDropDown()!;
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripMenuItem.cs:991]: ditto

**Review by RussKie — COMMENTED** [review]
👍

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripMenuItem.cs (hunk 2 above)
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/ToolStripMenuItem.cs:991]: Inside `IsDropDown` which is called before this line, there are some checks about the parent.

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**